astropy/convolution/boundary_extend.pyx

```diff
@@ -102,7 +102,7 @@ def convolve2d_boundary_extend(np.ndarray[DTYPE_t, ndim=2] f,
     cdef int wky = nky // 2
```

Comment:
You could make `wkx` and `wky` unsigned ints.  There's no reason these values will ever be negative.  I think by default Numpy array shapes are read as signed ints and I'm not sure why.  But if you cast these as unsigned ints then you don't have to worry about casting below.